scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,213 @@
+    if (_dropMode isEqualType [] && {count _dropMode == 3}) then {
+        _dropData = _dropMode;
+    };
+};
+
+_dropData params ["_dropHeight", "_dropLength", "_chuteHeight"];
+
+private _vehicle = vehicle (leader _pilotGroup);
+if(isNull _vehicle) exitWith {}; // group not in vehicle
+
+private _origHeight = (getPosATL _vehicle) select 2;
+_vehicle flyInHeight _dropHeight; // set forced flying height
+if (_resetHeight) then {
+    [{
+        params ["_vehicle"];
+        (fullCrew [_vehicle,"cargo"]) isEqualTo []
+    }, {
+        params ["_vehicle", "_origHeight"];
+        _vehicle flyInHeight _origHeight;
+    }, [_vehicle, _origHeight]] call CBA_fnc_waitUntilAndExecute;
+};
+
+// next process waypoints if waypoints desired. 
+// if waypoint is desired, then script will be run on setWaypointStatements
+// if not then script will be run in this function.
+
+// Use addWayPoint for the group piloting the vehicle to add the drop waypoints
+private _dropRunOrigin = getPosATL _vehicle; // calling position from leader will call last position on the ground
+
+// Obtain a random location within drop target
+// Between current position and drop target, find angle
+private _dropTarget = [_loc, true] call CBA_fnc_randPosArea;
+
+if (_dropTarget isEqualTo []) exitWith {};
+
+    diag_log [_pilotGroup];
+private _fnc_dzLocs = {
+    params ["_dropTarget","_dropRunOrigin","_dropLength"];
+    // getDir gives the azimuth, which differs in convention from trig
+    // in two ways: (1) azimuth has zero origin at North and
+    // (2) advances clockwise rather than counterclockwise
+    // To convert, use a conditional function:    
+    private _azimuth = _dropRunOrigin getDir _dropTarget;
+    private _theta = switch (floor _azimuth/90) do {
+        case 0: {90-_azimuth;};
+        default {450-_azimuth;};
+    };
+    // For a known drop point and known radius with trig angle _theta,
+    // and radius r given by _dropLength/2,
+    // then coordinates in the xy-plane relative to _dropTarget are:
+    // +/- r*cos _theta for x (Northing) and +/- r*sin _theta for x (Easting)
+
+    private _yDiff = (sin _theta) * _dropLength/2;
+    private _xDiff = (cos _theta) * _dropLength/2;
+    // set all coordinates based on DZ points in xy-plane; z coord does not matter
+    // double lengths on second DZ coordinate to ensure straight line flight
+    [[(_dropTarget select 0) + _xDiff,(_dropTarget select 1) + _yDiff,_dropHeight],
+        [(_dropTarget select 0) - _xDiff,(_dropTarget select 1) - _yDiff,_dropHeight]];
+};
+
+private _dropZoneLocs = [_dropTarget, _dropRunOrigin, _dropLength] call _fnc_dzLocs;
+private _minDistance = _dropRunOrigin distance (_dropZoneLocs select 1);
+
+// Handling for when calling waypoint is within distance _dropLength
+if (_minDistance < _dropLength) then {
+    _dropLength = _minDistance - 100; //arbitrary distance to allow for a helicopter turn
+    if (_dropLength < 0) exitWith {ERROR_1("Waypoint too close to DZ for drop!")};
```

Comment:
Actually, should this be an error? Or will this `exitWith` just exit the `if` statement above it and continue with the original `_dropZoneLocs`? If it shouldn't be an error, maybe a warning or entirely remove the message?

Reply:
You're right, even though `if` in the [docs](https://community.bistudio.com/wiki/exitWith) is not explicitly named as one of the local scopes that can be exited, it shows up in the examples.  Still, I think it should be an error, and the way to get it to properly be an error would actually be to code it as `exitWith {exitWith{ERROR_!("Waypoint too close to DZ for drop!")};};`